cmake/modules/hunter_get_source_subdir.cmake

```diff
@@ -0,0 +1,57 @@
+# Copyright (c) 2018, Ruslan Baratov
+# All rights reserved.
+
+include(CMakeParseArguments) # cmake_parse_arguments
+
+include(hunter_assert_empty_string)
+include(hunter_status_debug)
+include(hunter_assert_not_empty_string)
+
+function(hunter_get_source_subdir)
+  set(optional "")
+  set(one PACKAGE OUT)
+  set(multiple "")
+
+  # Introduce:
+  # * x_PACKAGE
+  # * x_OUT
+  cmake_parse_arguments(x "${optional}" "${one}" "${multiple}" "${ARGV}")
+
+  hunter_assert_empty_string("${x_UNPARSED_ARGUMENTS}")
+
+  hunter_assert_not_empty_string("${x_PACKAGE}")
+  hunter_assert_not_empty_string("${x_OUT}")
+
+  set(default_args "${__HUNTER_DEFAULT_SOURCE_SUBDIR_${x_PACKAGE}}")
+  set(user_args "${__HUNTER_FINAL_SOURCE_SUBDIR_${x_PACKAGE}}")
+
+  set(source_subdir "")
+
+  # Lowest priority: SOURCE_SUBDIR from 'hunter.cmake'
+  if(NOT default_args STREQUAL "")
+    hunter_status_debug(
+        "Package '${x_PACKAGE}' default arguments: '${default_args}'"
+    )
+    list(APPEND source_subdir "${default_args}")
+  endif()
+
+  # Priority is higher than default SOURCE_SUBDIR from 'hunter.cmake' but
+  # lower than user's SOURCE_SUBDIR from 'config.cmake'
+  if(NOT "${HUNTER_CACHED_BUILD_SHARED_LIBS}" STREQUAL "")
```

Comment:
What does `BUILD_SHARED_LIBS` have to do with anything here?

Reply:
copy wasted :see_no_evil: 

removed it, also using `set()` instead of `list(APPEND)` as `SOURCE_SUBDIR` is only one directory, not a list of directories